src/plugins/tscn/tscnplugin.cpp

```diff
@@ -454,6 +454,28 @@ static void writeTileset(const Map *map, QFileDevice *device, bool isExternal, A
                         device->write(formatByteString("%1/transpose = true\n", tileName));
 
                     foundCollisions |= exportTileCollisions(device, tile, tileName, alt);
+
+                    // Custom properties
+                    const Properties &properties = tile->properties();
+                    const auto keys = properties.keys();
```

Comment:
Instead of allocating an array for the keys we could iterate the properties directly by using:
```c++
QMapIterator<QString,QVariant> it(tile->properties());
while (it.hasNext()) {
    auto value = it.next().value();
    // access it.key() here
}
```
Also, we may want to use `tiled->resolvedProperties()`, if we also want to get the property values that the tile may have inherited from its class.